services/shhext/api.go

```diff
@@ -234,6 +252,102 @@ func waitForExpiredOrCompleted(requestID common.Hash, events chan whisper.Envelo
 	}
 }
 
+// RequestMessagesNew requests historic messages from a mail server
+// and blocks until the response from the mail server arrives.
+// nolint: gocyclo
+func (api *PublicAPI) RequestMessagesNew(ctx context.Context, r MessagesRequest) (MessagesResponse, error) {
+	log.Info("[shhext/PublicAPI.RequestMessagesNew] called", "request", r)
```

Comment:
not sure why code from RequestMessages needs to be copied. Isn't it enough to call RequestMessages and then wait for the event in a loop?

Reply:
Yeah, that's true. I though I would need to change more but apparently it is almost the same + waiting for an event.

Reply:
But I think changing the result of `RequestMessagesSync` is a better idea if no one uses it and I won't break compatibility with anyone.